staging/src/k8s.io/apiserver/pkg/server/filters/waitgroup.go

```diff
@@ -38,6 +38,7 @@ func WithWaitGroup(handler http.Handler, longRunning apirequest.LongRunningReque
 
 		if !longRunning(req, requestInfo) {
 			if err := wg.Add(1); err != nil {
+				w.Header().Add("Retry-After", "1")
 				http.Error(w, "apiserver is shutting down.", http.StatusInternalServerError)
```

Comment:
can we also switch this to a Status object?

Reply:
i would do this as a follow up (it might require more plumbing and there are more places in code that return plain text errors via `http.Error()`).

Reply:
is 1 appropriate?

Reply:
@smarterclayton my assumption is that HA clusters have 2 other endpoints the client-go can reconnect when it gets apiserver shutdown error. So faster retry will lead to faster success?